radix-engine/src/engine/call_frame.rs

```diff
@@ -171,30 +168,15 @@ pub enum StaticSNodeState {
     TransactionProcessor,
 }
 
-pub enum LoadedSNodeState<'a> {
+pub enum SNodeState<'a> {
     Static(StaticSNodeState),
-    Consumed(Option<ConsumedSNodeState>),
+    Consumed(ConsumedSNodeState),
     Borrowed(BorrowedSNodeState<'a>),
-}
-
-pub enum SNodeState<'a> {
-    Root,
-    SystemStatic,
-    TransactionProcessorStatic,
-    PackageStatic,
-    AuthZoneRef(&'a mut AuthZone),
-    WorktopRef(&'a mut Worktop),
-    // TODO: use reference to the package
-    Blueprint(ScryptoActorInfo, ValidatedPackage),
-    Component(ScryptoActorInfo, ValidatedPackage, &'a mut Component),
-    ResourceStatic,
-    ResourceRef(ResourceAddress, &'a mut ResourceManager),
-    BucketRef(BucketId, &'a mut Bucket),
-    Bucket(Bucket),
-    ProofRef(ProofId, &'a mut Proof),
-    Proof(Proof),
-    VaultRef(VaultId, &'a mut StoredValue),
-    TrackedVaultRef(VaultId, &'a mut Vault),
+    Tracked(
+        Address,
+        SubstateValue,
+        Option<(ScryptoActorInfo, ValidatedPackage)>,
```

Comment:
Looks like we should be able to merge `Address` with `ScryptoActorInfo`, and merge `SubstateValue` with `ValidatedPackage`.

Reply:
Yup looking to get this merged in future PRs :)